test/unit/puppetlabs/services/ca/certificate_authority_core_test.clj

```diff
@@ -119,6 +127,57 @@
           response (ring-app request)]
       (is (= version-number (get-in response [:headers "X-Puppet-Version"]))))))
 
+(deftest handle-delete-certificate-request!-test
+  (let [settings (assoc (testutils/ca-sandbox! cadir)
+                        :allow-duplicate-certs true
+                        :autosign false)]
+    (testing "successful csr deletion"
+      (logutils/with-test-logging
+        (let [subject "happy-agent"
+              csr-stream (gen-csr-input-stream! subject)
+              expected-path (ca/path-to-cert-request (:csrdir settings) subject)]
+          (try
+            (handle-put-certificate-request! subject csr-stream settings)
+            (is (true? (fs/exists? expected-path)))
+            (let [response (handle-delete-certificate-request! subject settings)
+                  msg-matcher (re-pattern (str "Deleted .* for " subject ".*"))]
+              (is (false? (fs/exists? expected-path)))
+              (is (= 204 (:status response)))
+              (is (re-matches msg-matcher (:body response)))
+              (is (= "text/plain" (get-in response [:headers "Content-Type"])))
+              (is (logged? msg-matcher :debug)))
+            (finally
+              (fs/delete expected-path))))))
+      (testing "Attempted deletion of a non-existant CSR"
+        (logutils/with-test-logging
+          (let [subject "not-found-agent"
+                response (handle-delete-certificate-request! subject settings)
+                expected-path (ca/path-to-cert-request (:csrdir settings) subject)
+                msg-matcher (re-pattern
+                              (str "No cert.*request for " subject " at.*" expected-path))]
+            (is (false? (fs/exists? expected-path)))
+            (is (= 404 (:status response)))
+            (is (re-matches msg-matcher (:body response)))
+            (is (= "text/plain" (get-in response [:headers "Content-Type"])))
+            (is (logged? msg-matcher :warn)))))
+      (testing "Error during deletion of a CSR"
+        (logutils/with-test-logging
+          (let [subject "err-agent"
+                csr-stream (gen-csr-input-stream! subject)
+                expected-path (ca/path-to-cert-request (:csrdir settings) subject)]
+            (try
+              (handle-put-certificate-request! subject csr-stream settings)
+              (is (true? (fs/exists? expected-path)))
+              (fs/chmod "-w" (fs/parent expected-path))
```

Comment:
yeah, I couldn't get it working by setting the permissions on the individual files and I was worried about setting the perms on a whole directory of test files. But Kevin said in another thread that it wasn't a concern since they were part a testing sandbox and to go for it.